k2/csrc/fsa_utils.cu

```diff
@@ -263,7 +263,7 @@ class OpenFstStreamReader {
     int32_t src_state, dest_state, label;
     line_is >> src_state >> dest_state >> label;
     if (!line_is.good()) {
```

Comment:
I think the actually problem may be here, in "if (!line_is.good()) {".
According to the table here, https://en.cppreference.com/w/cpp/io/ios_base/iostate, if eofbit is set but neither failbit nor badbit is set, good() will return false.  This is not what I intended.  The if-statement should probably read: "if (line_is.fail()) { .. }"